articles/seaborn-tutorial/index.md

```diff
@@ -0,0 +1,313 @@
+import matplotlib.pyplot as plt
+```
+
+Seaborn has 18 in-built datasets, which can be found using the following command.
+
+```python
+sns.get_dataset_names()
+```
+
+We will be using the Titanic dataset for this tutorial.
+
+```python
+df = sns.load_dataset('titanic')
+df.head()
+```
+
+---
+
+### Different Types of Graphs
+
+#### Count Plot
+
+A count plot is helpful when dealing with categorical values. It is used to plot the frequency of the different categories. The column **sex** contains categorical data in the titanic data, i.e., male and female.
+
+```python
+sns.countplot(x='sex',data=df)
+```
+![image title](/engineering-education/seaborn-tutorial/countplot1.png) 
+- **data** - The dataframe
+
+- **x** - The name of the column
+
+  
+
+We can further break up the bars in the count plot based on another categorical variable. The color palette of the plot can also be customized.
+
+```python
+sns.countplot(x='sex', hue = 'survived', data = df,
+palette = 'Set1')
+```
+![image title](/engineering-education/seaborn-tutorial/countplot2.png) 
+
+- **hue** - The name of the categorical column to split the bars
+
+- **palette** - The color palette to be used. For a list of color palettes, check out matplotlib's documentation [here](https://matplotlib.org/3.3.3/tutorials/colors/colormaps.html)
+  
+#### KDE Plot
+
+A Kernel Density Estimate (KDE) Plot is used to plot the distribution of continuous data.
+
+```python
+sns.kdeplot(x = 'age' , data = df , color = 'black')
+```
+![image title](/engineering-education/seaborn-tutorial/kdeplot.png) 
+
+- **data** - The dataframe
+
+- **x** - The name of the column
+
+- **color** - The color of the graph. You can find a list of colors [here](https://matplotlib.org/3.1.0/gallery/color/named_colors.html)
+
+  
+
+#### Distribution Plot
+
+A Distribution plot is similar to a KDE plot. It is used to plot the distribution of continuous data.
+
+```python
+sns.displot(x = 'age',kde=True,bins = 5 , data =df)
+```
+![image title](/engineering-education/seaborn-tutorial/distplot1.png)
+- **kde** - It is set to False by default. However, if you wish to plot a KDE graph on top of the bars, you can set it to True
+
+- **bins** - The number of bins/bars. The lower the number, wider the bards and wider the intervals
+
+```python
+sns.displot(x ='age',kde=True,bins = 5 ,
+hue = df['survived'] , palette = 'Set3', data=df)
+```
+![image title](/engineering-education/seaborn-tutorial/distplot2.png)
+
+You can also pass **hue** and **palette** as parameters to customize the graph.
+
+  
+
+#### Scatter Plot
+For this plot and the plots below, we will be working with the iris dataset. First, we will need to load the iris dataset.
+
+```python
+df = sns.load_dataset('iris')
+df.head()
+```
+Scatter plots help understand co-relation between data,
+
+```python
+sns.scatterplot(x='sepal_length', y ='petal_length' ,
+data = df , hue = 'species')
+```
+![image title](/engineering-education/seaborn-tutorial/scatterplot.png)
```

Comment:
What do I see here? How is the scatter plot agreeing with the iris dataset? If you can explain the plot and how to read those plots, this will be a great addition to the resources out there. Reading and understanding plots are equally important, a skill not that common.